mininterface/_tk_interface/utils.py

```diff
@@ -67,12 +69,55 @@ def get(self):
 
 
 def choose_file_handler(variable: Variable, tag: PathTag):
+    """Handler for file/directory selection on PathTag"""
     def _(*_):
-        if tag.multiple:
-            out = str(list(askopenfilenames(title="Choose files")))
+        initialdir = tag.val if tag.val else os.getcwd()
+
+        # Check whether this is a directory selection
+        if tag.is_dir:
+            # Directory selection using askdirectory
+            kwargs = {"title": "Select Directory", "initialdir": initialdir}
+            selected_dir = askdirectory(**kwargs)
+            if not selected_dir:  # User cancelled
+                return
+
+            if tag.multiple:
+                # Handle multiple selection for directories
+                current_dirs = tag.val if tag.val else []
+                if not isinstance(current_dirs, list):
+                    current_dirs = [current_dirs]
+                if selected_dir not in current_dirs:
+                    current_dirs.append(selected_dir)
```

Comment:
Use it like this – the Path is just cosmetics, the type checker raises a warning.

But the second line is important, without that, the input entry value would be with spaces. (Which might pose risk for slashing etc.) 

```py
                    current_dirs.append(Path(selected_dir))
                variable.set(str(list(str(x) for x in current_dirs)))
```